nussl/core/augmentation.py

```diff
@@ -0,0 +1,377 @@
+    
+    l_stft = audio_signal.stft_data.copy()
+    freq_vector = audio_signal.freq_vector
+    idx = (np.abs(freq_vector - highest_freq)).argmin()
+    l_stft[idx:, :, :] = 0
+    augmented_signal = audio_signal.make_copy_with_stft_data(l_stft)
+    augmented_signal.istft()
+
+    return augmented_signal
+
+def high_pass(audio_signal, lowest_freq):
+    """
+    Applies high pass filter
+    Args: 
+        audio_signal: An AudioSignal object
+        highest_freq: Threshold for high pass. Should be positive scalar
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(lowest_freq), np.number) or lowest_freq <= 0:
+        raise ValueError("lowest_freq should be positve scalar")
+
+    h_stft = audio_signal.stft_data.copy()
+    freq_vector = audio_signal.freq_vector
+    idx = (np.abs(freq_vector - lowest_freq)).argmin()
+    h_stft[:idx, :, :] = 0
+    augmented_signal = audio_signal.make_copy_with_stft_data(h_stft)
+    augmented_signal.istft()
+
+    return augmented_signal
+
+def tremolo(audio_signal, mod_freq, mod_depth):
+    """
+    https://ffmpeg.org/ffmpeg-all.html#tremolo
+
+    Applies tremolo filter
+    Args: 
+        audio_signal: An AudioSignal object
+        mod_freq: Modulation frequency. Must be between 0 and 1.
+        mod_depth: Modulation depth. Must be between 0 and 1.
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(mod_freq), np.number) or mod_freq < 0:
+        raise ValueError("mod_freq should be positve scalar")
+    
+    if not np.issubdtype(type(mod_depth), np.number) or mod_depth < 0 or mod_depth > 1:
+        raise ValueError("mod_depth should be positve scalar between 0 and 1.")
+
+    filter_kwargs = {
+        "f": mod_freq,
+        "d": mod_depth
+    }
+
+    return apply_ffmpeg_filter(audio_signal, "tremolo", **filter_kwargs)
+
+def vibrato(audio_signal, mod_freq, mod_depth):
+    """
+    https://ffmpeg.org/ffmpeg-all.html#vibrato
+
+    Applies vibrato filter
+    Args: 
+        audio_signal: An AudioSignal object
+        mod_freq: Modulation frequency. Must be between 0 and 1.
+        mod_depth: Modulation depth. Must be between 0 and 1.
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(mod_freq), np.number) or mod_freq < 0:
+        raise ValueError("mod_freq should be positve scalar")
+    
+    if not np.issubdtype(type(mod_depth), np.number) or mod_depth < 0 or mod_depth > 1:
+        raise ValueError("mod_depth should be positve scalar between 0 and 1.")
+
+    filter_kwargs = {
+        "f": mod_freq,
+        "d": mod_depth
+    }
+
+    return apply_ffmpeg_filter(audio_signal, "vibrato", **filter_kwargs)
+
+def chorus(audio_signal, in_gain=.4, out_gain=.4, delays=[50], 
```

Comment:
```suggestion
def chorus(audio_signal, in_gain=.4, out_gain=.4, delays=[50], 
decays=[.8], speeds=[.95], depths=[.7]):
```
Don't make default arguments mutable: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Simple fix here is to do `delays=None`, then take care of it in the body of the function `delays = delays if delays is not None else [50]`.